src/AdsImporter/Database/MongoMigration.php

```diff
@@ -137,19 +137,7 @@ public function addMessage(Message $message): void
             'length' => $message->getLength(),
         ];
 
-        try {
-            $this->messageCollection->insert($document);
-        } catch (\MongoDuplicateKeyException $ex) {
-            $details = sprintf(
-                'Id: %s, NodeId: %s, BlockId: %s, Length: %s',
-                $message->getId(),
-                $message->getNodeId(),
-                $message->getBlockId(),
-                $message->getLength()
-            );
-
-            $this->addExceptionToLog(sprintf('MESSAGE_DUPLICATED [%s]', $details));
-        }
+        $this->messageCollection->update(['_id' => $message->getId()], $document, ['upsert' => true]);
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "upsert" 5 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=php%3AS1192)